### ISSUE TITLE
Router: Middleware were already registered for the path

### ISSUE THREAD

**Issue by mikepc** [iris-contrib/middleware#14]
I have two paths that diverge only by VERB (following REST principals). 

The problem with the graceful package was fixed here https://github.com/kataras/iris/issues/198 

But the cors middleware still has the same problem

**Comment by ghiewa** [comment]
I ran into similar error, ask for help!

[IRIS] 2016/10/06 17:13:34 Error: Router: Middleware were already registered for
 the path: '/api/v0/items' !
panic: Error: Router: Middleware were already registered for the path: '/api/v0/
items' !

```
	iris.Post("/login", Login)
	iris.Post("/register", Register)

	iris.Post("/changepassword", Authenticated, ChangePassword)

	// Control Handler part
	iris.Post("/taskpost", Authenticated, PostTask)

	ap := iris.Party("/api/v0", func(ctx *iris.Context) {println("middleware here"); ctx.Next()})
	{
		//ap.UseFunc(Authenticated)   TODO  active while

		ap.API("/items", ItemAPI{})

	}
```

**Comment by kataras** [comment]
@mikepc @ghiewa Please try to have the same issue in the same post in order for me to answer one time per issue, no one time per person, here is the most  recent: https://github.com/kataras/iris/issues/461 ( 1 day before post your issue) the `198` was much more different case, for the `CORS` , wait for a fix soon!

**Comment by kataras** [comment]
@ghiewa  `.API` method, is not created to be used inside parties, this function is only for situtations who just want one API per thing, simple staff,  as I'm writing on examples and book .API is not created for production, it's not like 'controllers' iris is different type of framework, you have more 'low level' access.  However after I fix the CORS I will see your issue also, wait for fix here :)

**Comment by kataras** [comment]
@mikepc Thanks for the report, it's already written [here](https://github.com/kataras/iris/issues/461) one day before you post this, but no problem, just try to search before posting a new issue the next time please, we need to keep order to help new users find what they are searching for.

The issue with middleware already registered should be **fixed** with the v4.4.9, please upgrade the cors middleware ( github.com/iris-contrib/middleware/cors) and iris (github.com/kataras/iris) . Could you verify it, please?
